### PR TITLE
GLT-418 Add Targeted Resequencing to Library Dilutions

### PR DESCRIPTION
**Pull request by tonydebat**
Added db, dao and ui. Ui is available when displaying a single library from the list of libraries.

### CODE REVIEW

**Comment by djcooke** [comment]
:+1:

Why not Hibernate though?

**Comment by tonydebat** [comment]
I prefer hibernate, but wanted to avoid potential issues due to mixing the two schemes. At some point the entire persistence layer should be converted to hibernate in one go.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java [hunk 1]
```diff
@@ -2799,4 +2803,22 @@ public SequencerServiceRecord getSequencerServiceRecordById(long id) throws IOEx
   public Collection<LibraryPropagationRule> listLibraryPropagationRulesByClass(SampleClass sampleClass) throws IOException {
     return libraryPropagationRuleDao.getLibraryPropagationRulesByClass(sampleClass);
   }
+
+  @Override
+  public Collection<TargetedResequencing> listAllTargetedResequencing() throws IOException {
+    if (targetedResequencingStore != null) {
+      return targetedResequencingStore.listAll();
+    } else {
+      throw new IOException("No securityStore available. Check that it has been declared in the Spring config.");
```
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java]: s/security/targedResequencing/

File: miso-web/src/main/webapp/scripts/library_ajax.js [hunk 2]
```diff
@@ -263,17 +263,51 @@ Library.dilution = {
...
+    	          'libraryPrepKitId': libraryPrepKitId
+    	        },
+    	        {'doOnSuccess': function (json) {
+    	        	var selectElem = jQuery('#libraryDilutionTargetedResequencing');
+    	        	if(json.targetedResequencings.length == 0) {
+    	        		selectElem.append(new Option('NONE', 0));
+    	        	} else {
+    	        		selectElem.append(new Option('--- select'));
+    	        		selectElem.append(new Option('NONE', 0));
+        	        	jQuery.each(json.targetedResequencings, function(index, item) {
```
Comment (apmasell) [inline miso-web/src/main/webapp/scripts/library_ajax.js]: These should get sorted by name.

**Comment by apmasell** [comment]
:+1:

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java (hunk 1 above)
Comment (tonydebat) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java]: Updated.

File: miso-web/src/main/webapp/scripts/library_ajax.js (hunk 2 above)
Comment (tonydebat) [inline miso-web/src/main/webapp/scripts/library_ajax.js]: Added sort.

**Pull request merged**